src/mpyl/run_plan.py

```diff
@@ -81,3 +100,129 @@ def find_stage(plan: dict[Stage, set[ProjectExecution]]):
         if use_full_plan:
             return find_stage(self.full_plan)
         return find_stage(self.selected_plan)
+
+    def write_to_pickle_file(self):
+        logger = logging.getLogger("mpyl")
```

Comment:
I don't remember why, but we had a best practice of passing the logger as much as possible instead of creating/getting a new one all the time.

P.s. I'm assuming all of the below is only moved, not changed

Reply:
This code was indeed moved from a place where we already had a `logger` available. It's now called from a different place where we don't have a `logger` yet, and that's why I had to instantiate it here. The good news is that at least it's consistent with `write_to_json_file()` 😅 

In the near future I'd also love to align our usage of `Console` vs `Logger` vs `print()`, we use all three and it's extremely confusing to know when to use which one 🤷